spec/rubocop/cop/rspec/change_by_zero_spec.rb

```diff
@@ -25,54 +25,121 @@
+            .to change(Foo, :bar).by(0) &
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change(Foo, :baz).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0) &
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change { Foo.baz }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+
+    it 'registers an offense when the argument to `by` is zero ' \
+      'with compound expectations by `or`' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+            .or change(Foo, :baz).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+            .or change { Foo.baz }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+
+    it 'registers an offense when the argument to `by` is zero ' \
+      'with compound expectations by `|`' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0) |
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change(Foo, :baz).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0) |
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change { Foo.baz }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+  end
+
+  context "when `NegatedMatcher: 'not_change'`" do
+    let(:cop_config) { { 'NegatedMatcher' => 'not_change' } }
+
+    it 'registers an offense and autocorrect when ' \
+      'the argument to `by` is zero with compound expectations' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
```

Comment:
Does it make sense to tell about the defined negated matcher?
> Prefer `not_change` with compound expectations over `change.by(0)`

Reply:
Indeed it is.  We have made the following suggestions

> Prefer `not_change` with compound expectations over change.by(0)